README.md

```diff
@@ -25,4 +25,36 @@ Copy `.env.example` to `.env`, and run `docker-compose up`.
 - Lockfile now committed.
 - Configuration for dist building with `git archive`. Results in smaller dist packages.
 
+## Porting
+### Dhii to PSR-14
+The most common and generally intended course is that implementations of this standard will want to
+make themselves truly compatible with [PSR-14][] by implementing the FIG interfaces. This should be a very
+straight-forward process, as the interfaces in this package are designed to be fully compatible with
+PSR-14 interfaces. Therefore, all one would have to do is to make their dispatcher, listener, and
+stoppable event implementations _additionally_ implement their counterparts from the PSR-14 standrd.
+
+### PSR-14 to Dhii
+A less common direction would be to make implementations comply with the Dhii standard. If the reason
+is simply to support the standard, then the solution is very similar to the one above:
+simply make your implementations additionally implement Dhii interfaces. However, if the reason is
+to support earlier PHP versions (PSR-14 requires PHP >= 7.2), one would have to make a few additional
+changes to their code. Those changes are to the types accepted amd returned by your implementations.
+They can be found in the beginning of the _Details_ section above.
+
+The reason is that, while `Traversable` is a subtype of `iterable`, `array` is not a subtype of
+`Traversable`. Therefore, if your `ListenerProviderInterface` implementation returns arrays
+(which is valid with PSR-14), it must start returning `Traversable` instances instead. The simplest
+way to do this is by wrapping the array into an [`ArrayObject`][].
```

Comment:
I think [`ArrayIterator`](https://www.php.net/manual/en/class.arrayiterator.php) would be a better suggestion here.

Reply:
Could you elaborate on why that is so?

Reply:
The result should be a `Traversable`, and `ArrayIterator` simply wraps an array into an `Iterator`, making it the ideal minimal requirement. `ArrayObject` on the other hand implements a whole bunch of interfaces. Not to mention that it's technically not an `Iterator`, but an `IteratorAggregate`, therefore imposing a small overhead during iteration.

Reply:
![image](https://user-images.githubusercontent.com/1428973/72205238-035be100-3481-11ea-91dc-95d650bb1195.png)
Doesn't look so minimal to me.

Reply:
Wow I'm surprised. I guess I should have opened my own link before pasting :man_facepalming:

Though I still think there's an argument to be made here, as conceptually `ArrayObject` wraps arrays into an object while `ArrayIterator` into an `Iterator`. But I won't push my case anymore. PHP doing me a disappoint.

Reply:
I totally get the point you are trying to make. It sounds like it's just a iterator, and nothing more. But in reality, the only difference is the ability to seek, apparently.